data/scripts/sol-oic-gen.py

```diff
@@ -1220,7 +1220,7 @@ def master_c_as_string(generated, oic_gen_c, oic_gen_h):
     }
 
     SOL_INF("Found resource matching device_id");
-    resource->resource = sol_oic_resource_ref(oic_res);
+    resource->resource = sol_oic_resource_ref((struct sol_oic_resource *)oic_res);
```

Comment:
we could hide the counter in sol_oic_resource and make sol_oic_resource_ref receiving a const sol_oic_resource, so casts won't be necessary. It is hard for users to figure out when they can cast from const to non const.